documentation/processing.tex

```diff
@@ -26,16 +22,60 @@ \subsection{Overview}
 	\item \emph{Processing tracer} for tracing processed statements (see \cref{macro_tracer}).
 \end{itemize}
 
-\subsectionSrc{Types of statement processors}{parser\_library/src/processing/statement\_processors/statement\_processor.h}
-\label{lab06:sect_proc}
 
+\subsection{Composition}
+
+As the processing of the HLASM source file is rather complicated, we define a complex set of abstraction objects over the Gordian assembling of HLASM language:
+
+\begin{description}
+	\item[Statement provider] (see \cref{lab06:sect_prov}) is able to produce \TT{statement} structures. Its functionality is to provide statements from its finite statement stream.
+	
+	\item[Statement processor] (see \cref{lab06:sect_proc}) is an object that takes statement structures from a provider. Then, it performs a specific action with the acquired statement; namely, stores it into macro/copy definition (\emph{Macro/Copy processor}) or looks for sequence symbol (\emph{Lookahed processor}) or performs contained instruction (\emph{Opencode processor}).
+	
+	\item[Instruction processors] (see \cref{lab06:instr_proc}) help opencode statement processor in performing actions with the instructions contained in a statement. Each one of four instruction processors (Macro, Assembler, Machine and Conditional Assembly IP) process separate sub-set of a broad set of HLASM instructions. 
+	
+	\item[Instruction format validators] (see \cref{checker}) are helper objects used by instruction processors. They are provided operands of an instruction and serve to validate their correctness prior to the instruction. 
```

Comment:
helper objects mi evokuje nejakych par malych tried, nie najvecsiu komponentu co mame...

operands -> evaluated operands? 

prior to the instruction znamena podla mna "pred instrukciou", co mi nedava zmysel